x/zoneconcierge/keeper/grpc_query.go

```diff
@@ -72,6 +72,7 @@ func (k Keeper) ChainsInfo(c context.Context, req *types.QueryChainsInfoRequest)
 
 	ctx := sdk.UnwrapSDKContext(c)
 	var chainsInfo []*types.ChainInfo
+	// TODO: paginate this for loop
```

Comment:
Is the amount of data to be processed in this part likely to grow large enough to require paginating the loop? This is just a curiosity from a developer doing a code review!

Reply:
It is not clear how many integrations there will be yet. Though it does not hurt to always paginate for loops in the queries for avoiding DoS vectors